src/dbt_osmosis/core/osmosis.py

```diff
@@ -1029,6 +1030,31 @@ def remove_columns_not_in_database(
             )
         return changes_committed
 
+    @staticmethod
+    def get_prior_knowledge(
+        knowledge: Dict[str, Dict[str, Any]],
+        column: str,
+    ) -> Dict[str, Any]:
+        camel_column = re.sub("_(.)", lambda m: m.group(1).upper(), column)
+        prior_knowledge_candidates = [
+            knowledge.get(column),
+            knowledge.get(column.lower()),
+            knowledge.get(camel_column),
+        ]
+        sorted_prior_knowledge_candidates_sources = sorted(
+            [knowledge for knowledge in prior_knowledge_candidates if knowledge is not None and knowledge["progenitor"].startswith("source")],
```

Comment:
This for loop variable `knowledge` should not shadow the name of the local variable or else code flow is confusing.

Reply:
renamed in [9f66ac5](https://github.com/z3z1ma/dbt-osmosis/pull/89/commits/9f66ac50396ef84392994457c4049dc020be05a5) and [7dbbbb0](https://github.com/z3z1ma/dbt-osmosis/pull/89/commits/7dbbbb0e06b0df43119fb885a43e07f2fa9a7d26)